### DISCUSSION TITLE
Source/sink terms in percolation algorithms

### DISCUSSION THREAD

**Discussion by alper-can** [PMEAL/OpenPNM#1815] [Q&A]
Is it possible to give source/sink terms within percolation algorithms?

**Comment by jgostick** [comment]
Unfortunately, no.  Our percolation algorithms use graph theory to do percolation and are not physics based.  We did a [paper a few years ago](https://doi.org/10.1007/s10800-017-1126-6) where we looked at condensation and evaporation, but that involved 'flipping' back and forth between diffusion simulations to compute water vapor transport, then invasion percolation to inject or withdraw water based on the diffusion rate.

**Reply by alper-can** [reply]
Thank you very much for your answer. One more question I want to ask: How to couple diffusion simulation for water vapor transport and invasion percolation in the OpenPNM environment?

**Reply by jgostick** [reply]
The details are in the paper that I linked to.  It was not a pretty simulation and took ages to run.  My friend Pablo has recently published a paper that attempts to address this by using Ansys to harness the multiphysics.  The official paper is not showing up on his google scholar, but the preliminary work [is here](https://iopscience.iop.org/article/10.1149/09707.0615ecst/meta).
